### PR TITLE
improve detection of kitty image support

### PR DESCRIPTION
**Pull request by iambumblehead**
re [this comment](https://github.com/iambumblehead/render-thumb-for.sh/pull/14/files/09f0ec18e93932143fe4f5c7de666e41c27a961f#r1517930771) improve detection of kitty image support

### CODE REVIEW

**Comment by iambumblehead** [comment]
@xaizek feel free to review. I will smoke test now to verify the change is working here...

**Comment by iambumblehead** [comment]
```bash
echo "is_cmd_kitten_icat_support $is_cmd_kitten_icat_support"
```
kitty echoes "true" at the variable here and iterm does not echo anything there

**Comment by iambumblehead** [comment]
maybe this could be more compact but I'm not skilled enough with bash to know how
```bash
if [ -n "$is_cmd_kitten" ]; then
    if [ "$(kitten icat --detect-support 2>&1)" ]; then
        is_cmd_kitten_icat_support=true
    fi
fi
```

File: render-thumb-for.sh [hunk 1]
```diff
@@ -36,9 +36,8 @@ if [ -n "${XDG_CONFIG_HOME}" ]; then
 fi
 
 if [ -n "$is_cmd_kitten" ]; then
-    if [ ! "$(kitten icat --detect-support 2>&1)" ]; then
```
Comment (xaizek) [inline render-thumb-for.sh:39]: I think you meant:
```bash
if ! kitten icat --detect-support >/dev/null 2>&1; then
```

Comment (iambumblehead) [reply render-thumb-for.sh:39]: maybe I'm doing something wrong but this condition never evaluates as true, even from kitty
```bash
if ! kitten icat --detect-support >/dev/null 2>&1; then
    is_cmd_kitten_icat_support=true
fi
```

this expression shows "memory" in the in the shell, printed by the --detect-support command, but does evaluate true in kitty and false in iTerm2
```bash
if kitten icat --detect-support 2>&1; then
    is_cmd_kitten_icat_support=true
fi
```

this expression does not show "memory" and evaulates true in kitty, falce in iTerm2
```bash
if [ $(kitten icat --detect-support 2>&1) ]; then
    is_cmd_kitten_icat_support=true
fi
```

**Comment by iambumblehead** [comment]
it is changed to this
```bash
[ $(command -v kitten) ] &&
    is_cmd_kitten_icat_support=$(kitten icat --detect-support 2>&1)
```

**Comment by iambumblehead** [comment]
shellcheck fails with warning when the dollar sign expression is not wrapped in double-quotes, so this is what is there now
```bash
[ "$is_cmd_kitten" ] &&
    is_cmd_kitten_icat_support=$(kitten icat --detect-support 2>&1)
```

**Comment by iambumblehead** [comment]
I could make follow-up PRs... but let's merge this now so I can make a few changes for xterm with no outstanding branches

**Pull request merged**